File: GeoBroker-Server/src/test/java/de/hasenburg/geobroker/server/distribution/BrokerInfoTest.java
package de.hasenburg.geobroker.server.distribution;

import de.hasenburg.geobroker.commons.model.BrokerInfo;
import de.hasenburg.geobroker.commons.model.JSONable;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

@SuppressWarnings({"OptionalGetWithoutIsPresent"})
public class BrokerInfoTest {

	private static final Logger logger = LogManager.getLogger();

	@Test
	public void testSerialize() {
		BrokerInfo brokerInfo1 = new BrokerInfo("brokerId", "address", 1000);
		String json = JSONable.toJSON(brokerInfo1);
		logger.info(json);
		BrokerInfo brokerInfo2 = JSONable.fromJSON(json, BrokerInfo.class).get();
		assertEquals(brokerInfo1, brokerInfo2);
	}

}

File: GeoBroker-Commons/src/test/kotlin/de/hasenburg/geobroker/commons/model/disgb/BrokerInfoTest.kt
package de.hasenburg.geobroker.commons.model.disgb

import org.apache.logging.log4j.LogManager
import org.junit.Assert
import org.junit.Test
import java.lang.Exception

private val logger = LogManager.getLogger()

class BrokerInfoTest {

    @Test
    fun testSerialize() {
        val brokerInfo1 = BrokerInfo("brokerId", "address", 1000)
        val json1 = brokerInfo1.toJson()
        logger.info(json1)
        val brokerInfo2 = json1.toBrokerInfo()
        Assert.assertEquals(brokerInfo1, brokerInfo2)
    }

    @Test(expected = Exception::class)
    fun testFailedSerialize() {
        val brokerInfo = BrokerInfo("brokerId", "address", 1000)
        val json = brokerInfo.toJson()
        val wrong = json.toBrokerArea()
    }

}